### PR TITLE
docs(resources): document default resource requests

### PR DESCRIPTION
**Pull request by tthvo**
Related to #437 

Updated config docs for default resource requests. And fix minor typos.

### CODE REVIEW

**Comment by tthvo** [comment]
Sorry, just realized I haven't updated docs after #437, so I did it now.

File: docs/config.md [hunk 1]
```diff
@@ -188,6 +188,8 @@ spec:
 ```
 This example sets CPU and memory requests and limits for each container, but you may choose to define any combination of requests and limits that suits your use case.
 
+Note that if you define limits lower than default requests, the limits will be used in place of requests instead.
```
[the remark this answers is not in the document]
Comment (tthvo) [reply docs/config.md:191]: Oh yeh this is clearer to understand :D I will fix this.

File: docs/config.md [hunk 2]
```diff
@@ -152,8 +152,27 @@ spec:
     subProcessMaxHeapSize: 200
 ```
 
+If the sidecar's resource requests are not specified, they are set with the following defaults:
+
+| Request | Quantity |
+|---------|----------|
+| defaultReportCpuRequest | 128m |
```
Comment (ebaron) [inline docs/config.md:159]: I think it's probably better to specify this in plain English instead of using the constant names. Something like "Reports Container CPU/Memory".

File: docs/config.md [hunk 3]
```diff
@@ -152,8 +152,27 @@ spec:
...
+By default, the operator deploys Cryostat with pre-configured resource requests:
+
+| Request | Quantity |
+|---------|----------|
+| defaultCoreCpuRequest | 100m |
+| defaultCoreMemoryRequest | 384Mi |
+| defaultJfrDatasourceCpuRequest | 100m |
+|	defaultJfrDatasourceMemoryRequest | 512Mi |
+|	defaultGrafanaCpuRequest | 1000m |
+|	defaultGrafanaMemoryRequest | 256Mi |
```
Comment (ebaron) [inline docs/config.md:173]: Same thing here. "Cryostat container CPU/memory", "JFR Data Source container CPU/memory", "Grafana container CPU/memory".

File: docs/config.md (hunk 2 above)
Comment (tthvo) [reply docs/config.md:159]: Right! I will change that!

**Review by ebaron — APPROVED** [review]
Looks good!
